lib/db/mysql/index.js

```diff
@@ -2,85 +2,103 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+const path = require('path');
 const mysql = require('mysql');
 const buf = require('buf').hex;
+const MysqlPatcher = require('mysql-patcher');
 
-const AppError = require('../error');
-const config = require('../config');
-const logger = require('../logging')('db.mysql');
-const P = require('../promise');
+const AppError = require('../../error');
+const config = require('../../config');
+const logger = require('../../logging')('db.mysql');
+const P = require('../../promise');
+const patch = require('./patch');
 
-const SCHEMA = require('fs').readFileSync(__dirname + '/schema.sql').toString();
 
 function MysqlStore(options) {
   if (options.charset && options.charset !== 'UTF8_UNICODE_CI') {
-    logger.warn('createDatabase', { charset:  options.charset });
+    logger.warn('createDatabase', { charset: options.charset });
   } else {
     options.charset = 'UTF8_UNICODE_CI';
   }
+  options.typeCast = function(field, next) {
+    if (field.type === 'TINY' && field.length === 1) {
+      return field.string() === '1';
+    }
+    return next();
+  };
```

Comment:
It's passed an option to `mysql.createPool`. By default, `mysql` treats `BOOLEAN` columns as `TINY`, and returns `1` or `0`. That works in most cases, when just testing truthiness, but it can cause problems when testing identity. So this function simply evaluates the field into a boolean instead.